components/motor/gpio/motor_encoder_test.go

```diff
@@ -169,7 +162,9 @@ func TestEncodedMotor(t *testing.T) {
 	})
 
 	t.Run("encoded motor test GoFor forward", func(t *testing.T) {
-		test.That(t, m.goForInternal(context.Background(), 10, 1, 1), test.ShouldBeNil)
+		go func() {
+			test.That(t, m.GoFor(context.Background(), 10, 1, nil), test.ShouldBeNil)
```

Comment:
can you explain why GoFor rather than goForInternal? I don't see a major difference just wondering why the switch

Reply:
I'd rather test the full flow of the API than the internal helper function. We don't do this much at viam, but usually, you call your test files `package_tests` and then you can only test public functions and sturcts.